### DISCUSSION TITLE
samples: net: W5500 implementation

### DISCUSSION THREAD

**Discussion by yaadc** [zephyrproject-rtos/zephyr#46145] [General]
**Describe the bug**
The W5500 driver documentation looks quite vague, maybe it is just me.
Is there a chance to include a simple example showing how to use the W5500 driver?

Thanks.

**Comment by mbolivar-nordic** [comment]
@rlubos any advice here?

**Comment by mbolivar-nordic** [comment]
Converting to discussion

**Comment by rlubos** [comment]
Indeed it seems there is no example usage of this driver, other than a generic Ethernet buildall test, it doesn't also seem to be enabled automatically by any board/shield. That is a pity, I don't have the hardware though to verify the driver and add support to one of the samples. Perhaps the driver author @parthitce could help here?

**Reply by parthitce** [reply]
Development was done with a custom breakout board and consumed in customer hardware in my end. Do we need specific sample application for ethernet drivers? Is it not all the generic samples under samples/net should work by enabling under Kconfig?

Let me know if it makes sense to add device tree overlay otherwise to nrf52 or frdm kind.

**Reply by rlubos** [reply]
I didn't mean a specific sample for ethernet driver, but rather enabling the driver in one of the existing samples (`zperf` for instance) by adding driver-specific configuration to the sample.

Usually, it's not needed, as particular drivers are enabled automatically by selecting the appropriate board/shield - that way the board/shield is responsible for providing a valid configuration for the driver. However in this case, the driver is not associated with any board/shield, so it can't be selected automatically. We'd need an overlay that'd show how to enable the driver in an application.

**Reply by parthitce** [reply]
I understand. Will add a shield for [this](https://www.amazon.de/-/en/Ethernet-Network-Hardware-Microcontroller-Program/dp/B07JLFN3T1/ref=sr_1_3?crid=2KTPTO2GPN0CY&keywords=w5500&qid=1654076954&sprefix=w5500%2Caps%2C71&sr=8-3) breakout board for reference development. Hope that works?

**Reply by rlubos** [reply]
From my perspective, this should be fine. Thanks!

**Reply by parthitce** [reply]
Opened it here: https://github.com/zephyrproject-rtos/zephyr/pull/46201
